
Archived entries from file /Users/ian/pessoal/essencial/agenda.org

* 2022

** 2022-01 January

*** 2022-01-14 Friday
**** DONE Prepare Turkana repo [4/4]
CLOSED: [2022-01-14 Fri 18:21] DEADLINE: <2022-01-21 Fri>
:PROPERTIES:
:ARCHIVE_TIME: 2022-01-14 Fri 18:21
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection
:END:
- State "DONE"       from "TODO"       [2022-01-14 Fri 18:21]
- [X] Create =/fs/cbsuclarkfs1/storage/ivc2/turkana=.
- [X] Make the local folder a git and github repo.
- [X] Clone repo to =~/turkana= in the cluster.
- [X] Set up an archival document locally for Turkana todos.

*** 2022-01-17 Monday
**** DONE Plot Turkana SFS [10/10]
CLOSED: [2022-01-17 Mon 17:07]
:PROPERTIES:
:ARCHIVE_TIME: 2022-01-24 Mon 09:41
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana first simulations
:END:

- State "DONE"       from "DOING"      [2022-01-17 Mon 17:07]
- State "DOING"      from "TODO"       [2022-01-17 Mon 16:23]
- [X] Prepare snakefile for preparing the raw data.
- [X] Rule: concatenate SFS raw data files and convert into proper SFS.
- [X] Prepare snakefile for plotting in R.
  - [X] Look up access to snakemake variables in an R notebook
  - [X] Create conda environment for R notebooks
- [X] Rule: plot proper SFS together with a 1/n SFS to see if it fits neutrality.
  - [X] Look up the 1/n plotting in Ronen et al., 2013
  - [X] Normalize raw SFS
  - [X] Add reference for 1/x SFS
  - [X] Color, labels, prettify, etc.

*** 2022-01-19 Wednesday
**** DONE Turkana first simulations, simple parameters [39/39]
CLOSED: [2022-01-19 Wed 14:33]
:PROPERTIES:
:ARCHIVE_TIME: 2022-01-24 Mon 09:41
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana first simulations
:END:
- State "DONE"       from "DOING"      [2022-01-19 Wed 14:33]
- State "DOING"      from "TODO"       [2022-01-18 Tue 15:49]
- [X] Set up simulations locally
  - [X] Same set of scripts, etc. as the Drosophila project.
  - [X] Set up easy to use neutral parameters
  - [X] Run locally to test the configs
- [X] Simulation parameters
  - [X] Constant Ne of 10,000.
  - [X] Look up size of locus in my notes and in the raw empirical data
  - [X] Look up human mutation and recombination rate in SLiM manual.
  - [X] Change =project_parameters=: size of locus, smallest window size.
  - [X] Pi normalization
    - [X] Data preparation rule with =vcftools= windowed pi of empirical data.
    - [X] What's the highest value? Use that as normalization.
  - [X] Number of SNPs normalization: use total number in empirical VCF
  - [X] Test local simulations with these parameters. Note run time, window sizes.
- [X] Make a simple hard sweep config, with s between 0.1% and 100%. That's 3 orders of magnitude
- [X] Use recapitation instead of a coalescent burn-in.
  - [X] Git update everything, so all files are backed up nicely.
  - [X] No rule for burn-in, burn-in not required.
  - [X] Slim templates don't read burn-ins anymore.
  - [X] Slim templates get a population size parameter and initialize their subpops.
  - [X] Test run it now.
  - [X] Look up recapitation recipe.
  - [X] Add recapitation to mutation dropping script.
  - [X] Test run it again.
  - [X] Change name of mutation dropping everywhere to recapitation and mutation dropping
  - [X] Update git with a new branch
  - [X] Try updating SLiM to 3.7, see if that removes the warning we're getting
- [X] Set up cluster simulations
  - [X] Update slim on cluster to 3.7, add it to =~/bin/slim3.7=
  - [X] Fetch and merge repo on cluster
  - [X] Same set of scripts, etc. as the Drosophila project
  - [X] Edit slurm script
  - [X] Edit slurm submission script: can do more simulations per job, now
  - [X] Create conda repo on cluster using snakemake
  - [X] Debug cluster simulations
  - [X] Clean cluster =raw-simulations= folder
  - [X] Run neutral and 80% (ongoing) (hard) sweep simulations to start as a test, with 5000 simulations
  - [X] Update git from cluster
  - [X] Update local repo

*** 2022-01-20 Thursday
**** DONE Finalize Turkana first simulations [2/2]
CLOSED: [2022-01-20 Thu 17:31]
:PROPERTIES:
:ARCHIVE_TIME: 2022-01-24 Mon 09:41
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana first simulations
:END:
- State "DONE"       from "TODO"       [2022-01-20 Thu 17:31]
- [X] Confirm simulations ran okay
- [X] Download simulations locally, to same directory as I did with Drosophila

*** 2022-01-24 Monday
**** DONE Better Turkana SFS [11/11]
CLOSED: [2022-01-24 Mon 17:28]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana simple power analysis
:END:

- State "DONE"       from "DOING"      [2022-01-24 Mon 17:28]
- State "DOING"      from "TODO"       [2022-01-24 Mon 13:38]
- Are there batch effects in the intermediate peaks? Those intermediate peaks are worrying.
- Is there exponential growth?
- We don't expect a lot of sequencing errors in the excess of singletons; technology is pretty alright these days with that.
- Excess of high-freq. derived: misidentification of ancestral state?
- [X] Look up neutral expectation for folded spectrum. Philipp's lecture notes?
- [X] Rewrite SFS code to plot folded and unfolded in the same plot.
- [X] x-axis is allele frequency.
- [X] Add neutral expectation as a dashed black line.
- [X] Plot again, with better y-axis.
- [X] Re-name column of SFS data with =num_alternate=.
- [X] Re-plot a folded SFS.
- [X] We can use Fu (1995) to get the expected folded SFS under neutrality. Estimate \theta first from the first class of data, then see how it decays. Need to work it out on paper: what's \eta_i/\eta_1?
- [X] Plot a scaled, folded SFS.
- [X] Are there still weird peaks?
- [X] E-mail Philipp.

*** 2022-01-25 Tuesday
**** DONE Plan next steps of Turkana simulation [3/3]            :network:
CLOSED: [2022-01-25 Tue 15:35]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana simple power analysis
:END:

- State "DONE"       from "DOING"      [2022-01-25 Tue 15:35]
- State "DOING"      from "TODO"       [2022-01-25 Tue 15:17]
- [X] Look up the spelling of Julien's name
- [X] How to do recapitation with custom coalescence?
- [X] Reply to Philipp's email.

*** 2022-01-27 Thursday
**** DONE Simulate Turkana with demography [27/27]
CLOSED: [2022-01-27 Thu 13:30]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana simple power analysis
:END:

- State "DONE"       from "DOING"      [2022-01-27 Thu 13:30]
- State "DOING"      from "DONE"       [2022-01-27 Thu 13:27]
- State "DONE"       from "DOING"      [2022-01-26 Wed 17:08]
- State "DOING"      from "TODO"       [2022-01-25 Tue 15:44]
- [X] Do pi and SNP limit in 100kb empirical windows for feature normalization.
- [X] Implement demographic model in recapitation.
  - [X] Look up how =msprime= measures time.
  - [X] Write up a =demography= object with the correct parameters. Put it in the recapitation script.
  - [X] Double-check how to add demography to recapitation.
  - [X] Turn everything into parameters.
  - [X] Confirm parameters from document we have.
  - [X] Recapitate with that demography.
- [X] Update local parameters:
  - [X] Increase smallest window size to 10kb, and/or change number of windows from 21 to 11.
  - [X] Keep mutation and recombination rates at 1e-8.
  - [X] Pi normalization limit: 0.0025, kind of ad hoc based on 10kb empirical windows. Change it to 0.003, probably.
  - [X] SNP limit for normalization is at 200, changed to 500.
- [X] Change current population size to 30,000.
- [X] Test run simulations locally.
- [X] Add rule to compress .ms files.
  - [X] Function to get the inputs.
  - [X] Rule itself, based on the features one.
  - [X] Add requirement to the =all= rule.
- [X] Save those compressed files on the cluster.
- [X] Test it locally. (How long to compress ms?)
- [X] Make sure you can untar the ms
- [X] Git update locally and on cluster.
- [X] Clear folder of SLURM outputs on the cluster.
- [X] Send simulations out to the cluster.
- [X] Update git locally and on the cluster.
- [X] Re-send simulations to the cluster with 13 dimensions instead of 11.

*** 2022-01-29 Saturday
**** DONE Contact Amanda and Julien about the SFS [2/2]
CLOSED: [2022-01-29 Sat 10:15]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection
:END:

- State "DONE"       from "DOING"      [2022-01-29 Sat 10:15]
- State "DOING"      from "TODO"       [2022-01-28 Fri 14:14]
Are you aware of this? It's not binning. Do you have an explanation of it? Otherwise, I just want to point out the observation.
- [X] Write the email, ask Philipp about it
- [X] Correct it based on Philipp's comments, send it out.

Hi all,

I wanted to bring to the group an observation about the Turkana site-frequency spectrum. I plotted it as a folded SFS with unbinned frequencies on the x-axis, and the y-axis is plotted in relation to the expected neutral folded SFS. The empirical data has a conspicuous excess of variants with minor allele frequency around 46/220 samples and at frequency 108/220. I wanted to know if you had seen this pattern in the data before and if you have an idea of what could be causing it...

Best wishes,
Ian

***** DONE Perform simple Turkana power analysis [14/14]
CLOSED: [2022-01-28 Fri 12:19]
- State "DONE"       from "DOING"      [2022-01-28 Fri 12:19]
- State "DOING"      from "TODO"       [2022-01-27 Thu 11:44]
- [X] Wait for simulations to finish running.
- [X] Download simulations.
- [X] Copy deep learning notebooks from the /Drosophila/ repo.
- [X] Update snakefile rules for machine learning.
- [X] Update config file for machine learning.
- [X] Add conda env for machine learning.
- [X] Run all the rules, adding the scripts and notebooks as necessary.
- [X] Debug neural network dimension mismatch
- [X] Run snakemake, train models.
- [X] Look at the data info: does it make sense?
- [X] Re-run with 25 epochs of training.
- [X] Look at the fitting report of selection strength. We want good performance.
- [X] Git update.
- [X] Do a janky R notebook to plot the train/valid true vs. predicted selection strength.

** 2022-02 February

*** 2022-02-01 Tuesday
**** DONE Turkana better simulations parameters [18/18]
CLOSED: [2022-02-01 Tue 14:40]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana power analysis of ongoing sweep models
:END:
- State "DONE"       from "DOING"      [2022-02-01 Tue 14:40]
- State "DOING"      from "TODO"       [2022-02-01 Tue 12:13]
- State "SOON"       from              [2022-01-31 Mon 18:13]
- [X] Look up how to calculate \pi from the SFS in Philipp's lecture notes.
- [X] Derive \pi from the empirical SFS we have.
  - [X] Make a rule in data preparation with inputs and outputs.
  - [X] Make a vector of h_k per SFS entry.
  - [X] Multiply that by the counts of each SFS entry.
  - [X] Divide that by the total genome size of 3x10^9.
  - [X] Email Philipp asking if dividing by 3x10^9 is the right number.
- [X] Use N_e = 30,000 and derive \mu from that empirical \pi. Recombination rate then should be equal to \pi so \pi/r = 1.
- [X] RNM parameters: \theta_A can go from 0.1 (0.01?) to 5, or something like that.
- [X] SGV parameters: f_0 can go from 2/2Ne to 5% (10%? If it's feasible).
- [X] Recapitation demography needs to happen 7000 - (sweep time) generations ago. This matters for slow sweeps.
- [X] Record sweep origin generation to compute sweep age.
  - [X] SGV suggestion: generation where the allele became adaptive.
  - [X] RNM suggestion: age of the oldest mutation.
- [X] Test it all locally.
- [X] Git update.
- [X] Empty cluster directory of slurm outputs.
- [X] Send simulations to the cluster.

*** 2022-02-03 Thursday
**** DONE Plot SFS of simulated neutral demographic model [10/10]
CLOSED: [2022-02-03 Thu 16:23]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana power analysis of ongoing sweep models
:END:
- State "DONE"       from "DOING"      [2022-02-03 Thu 16:23]
- State "DOING"      from "SOON"       [2022-02-03 Thu 14:59]
- State "SOON"       from "TODO"       [2022-01-31 Mon 21:52]
- State "DOING"      from "TODO"       [2022-01-31 Mon 21:50]
- [X] Make rule and notebook to extract SFS in Python
- [X] Number of neutral simulations to use is a parameter
- [X] Wait for /neutral/ simulations to finish running on the cluster.
- [X] Download simulations
- [X] Use =tarfile= to extract the .ms files
- [X] Look up in the previous notebooks how to best read in genotype data from an .ms file
- [X] The neutral SFS can be obtained with a simple =Counter=, I am pretty sure
- [X] Adjust notebook of plotting SFS to also plot the neutral simulated one (adjust rule input too)
- [X] We expect neutral SFS to have an excess of neutral variants, but not as much excess as the genome-wide empirical data
- [X] Git update

**** DONE Turkana machine learning models [7/7]
CLOSED: [2022-02-03 Thu 18:13]
:PROPERTIES:
:ARCHIVE_TIME: 2022-02-14 Mon 09:52
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_ITAGS: project
:ARCHIVE_OLPATH: PhD/Turkana selection/Turkana power analysis of ongoing sweep models
:END:
- State "DONE"       from "DOING"      [2022-02-03 Thu 18:13]
- State "DOING"      from "SOON"       [2022-02-03 Thu 16:53]
- State "SOON"       from              [2022-01-31 Mon 18:13]
  - [X] Acquire empirical windows.
  - [X] Run model training with 1 epoch to test.
  - [X] Consolidate data, check how many sweeps we're training on.
  - [X] Implement rules for all 4 targets.
  - [X] Run model training with 25 epochs.
  - [X] What is the power we get?
  - [X] What are the inference results on the empirical data?

*** 2022-02-18 Friday
**** DONE Turkana, further prediction models [4/4]
CLOSED: [2022-02-18 Fri 15:45] DEADLINE: <2022-02-18 Fri>
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-03 Thu 18:25
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_OLPATH: PhD/Turkana selection
:END:

- State "DONE"       from              [2022-02-18 Fri 15:45]
***** DONE Turkana simulations [15/15]
CLOSED: [2022-02-16 Wed 22:48]
- State "DONE"       from "DOING"      [2022-02-16 Wed 22:48]
- State "DOING"      from "TODO"       [2022-02-14 Mon 15:52]
- Better recombination rate, maybe there's a good estimate from a different population out there. A genome browser or HapMap or something else - ask someone about the state of the art for human recombination rate?
- 10,000 copies of each simulation mode. Circa 60% of SGV and RNM simulations will succeed in total, so run 20,000 of them.
- [X] Ask someone about state-of-the-art human recombination rate estimates.
- [X] Find a recombination rate for our region: chromosome 8, the STC1 coordinates.
- [X] Add a source website for the recombination map
- [X] Make rule to get recombination rate
- [X] Open notebook from rule
- [X] Get coordinates of the window of interest: chr. 8; 23,350,029 - 24,424,864.
- [X] How to derive recombination rate from the maps?
- [X] Get one rate for each map
- [X] Plot for recombination across chromosome 8, add a line indicating where the region is.
- [X] Confirm: what mu/r does it produce?
- [X] Test run simulations locally with those parameters to see how they benchmark and how that compares to previous parameters.
- [X] Set up cluster scripts to run 20,000 simulations of SGV and RNM, and 10,000 hard sweeps, and 1,000 neutral simulations.
- [X] Ask Elissa about the limitation of 10,000 array jobs and slurm error I was getting: =sbatch: error: Slurm temporarily unable to accept job, sleeping and retrying=.
- [X] Change script to send only 5000 simulations of each kind.
- [X] Update git and run all.
  
***** DONE Turkana results with higher recombination rate
CLOSED: [2022-02-17 Thu 21:39]
- State "DONE"       from "DOING"      [2022-02-17 Thu 21:39]
- State "DOING"      from "DONE"       [2022-02-17 Thu 21:33]
- [X] Figure out units of the recombination maps
- [X] Correct units in notebook.
- [X] Produce a fine-scale map, too. Add that to the same table.
- [X] We want a fine scale plot to see hotspots of ~1kb
- [X] A larger scale map to see if the sweep 1Mb is an outlier among other 1Mb regions
- [X] Delete simulations with higher recombination
- [X] Send out more simulations, in batches of 1000 each.
  
***** DONE Turkana machine learning targets [18/18]
CLOSED: [2022-02-17 Thu 22:15]

- State "DONE"       from "TODO"       [2022-02-17 Thu 22:15]
- State "SOON"       from              [2022-02-07 Mon 10:59]
- Philipp's idea is to infer sweep mode first (with possibilities of the various patial sweep models) and then tailor the simulation parameters for training of the next models using only sweeps of the inferred mode. We should have a lot of power for everything since the partial sweep frequency is circa 80%.
- [X] What do I need to predict a new target? Write it down here as a todo for the following targets:
- [X] Predict SGV initial frequency without worrying about partial sweep models yet! The hope is that we're not getting to top end of our f_0 prior (currently 10%).
  - [X] Add target to config
  - [X] Target column in =utils.prepare_data=
  - [X] Balancing function in =utils.prepare_data=
- [X] What is the sweep age?
  - [X] Add target to config
  - [X] Target column in =utils.prepare_data=
  - [X] Balancing function in =utils.prepare_data=
  - [X] Target column in model application notebook
- [X] Infer # of adaptive alleles contributing to fixation for RNM sweep.
  - [X] Add target to config
  - [X] Target column in =utils.prepare_data=
  - [X] Balancing function in =utils.prepare_data=
- [X] Run everything with =-n=; do all the targets work?
- [X] Fit all models with 1 epoch.
- [X] Git update.
- [X] Fit all models with 50 epochs.
  
***** DONE Turn Philipp's ideas into more Turkana to-do's
CLOSED: [2022-02-18 Fri 15:18]

- State "DONE"       from "DOING"      [2022-02-18 Fri 15:18]
- State "DOING"      from "TODO"       [2022-02-18 Fri 11:38]
- [X] Update PhD plan with the new Turkana timeline.
- [X] Reply to Amanda.
- We've been using the demog. inferences from the analysis, and that's worked well for us so far. Should we not be using them? Is the demographic analysis still part of the project?

*** 2022-02-21 Monday
**** DONE Turkana sanity simulations
CLOSED: [2022-02-21 Mon 15:51]
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-09 Wed 12:27
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_OLPATH: PhD/Turkana project/Turkana, more confident predictions
:END:
- State "DONE"       from "DOING"      [2022-02-21 Mon 15:51]
- State "DOING"      from "SOON"       [2022-02-21 Mon 14:34]
- State "SOON"       from              [2022-02-21 Mon 11:52]
- [X] Git pull locally.
- [X] Prepare simulations with dimension=21 to see if that makes a difference.
- [X] What needs to change about RNM sweep dominance?
- [X] Prepare simulations with h=1, it will be a robustness test.
  - [X] Run local simulations to generate SLiM scripts.
  - [X] Use the specified dominance coefficient.
  - [X] Callback: if the other genome doesn't have a mutation of same type, use the dominance coefficient.
  - [X] It it does, use the special formula for h.
  - [X] Try to debug this in slimgui somehow.
  - [X] Change slim script template.
  - [X] Change parameter instantiation script.
- [X] Test run locally.
- [X] Git update on the cluster.
- [X] Send jobs out.
  

** 2022-03 March

*** 2022-03-03 Thursday
**** Project description
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-03 Thu 18:25
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_OLPATH: PhD/Turkana selection/Meetings
:END:

- Paper is a story about adaptation to desert living and dehydration with the STC1 locus.
- The locus is also "super pleiotropic".
- Paper will introduce entire dataset of 301 Turkana genomes.
- Original purpose of the first 100 high-quality Turkana genomes was to compare cultural diverstiy with genetic diversity (cultural F_ST vs. genetic F_ST), but that won't go in this paper. Another group is working on it.
- Paper includes selection scan, functional validation, and hopefully some inferences about the mode and strength of selection. That's where we come in.
- The sweep in STC1 with circa 80% frequency.
- What is selection strength?
- Is the sweep hard or soft? This is an interesting question.
- Estimate sweep age if it's straightforward, but not necessary if it's a lot of trouble. I think it's a lot of trouble because it'd require assuming some model for a sweep that's both incomplete and old.
- The candidate region for the sweep is around 50kb, and we can try some different 50kb windows and see if results are different.
- Out timeline is weeks to months.
- What demographic model to simulate? Try equilibrium first. They have a two-epoch expansion model fit with dadi, with N_e of 15,000 and 30,000.
- Those population sizes are great because they don't require rescaling.
- Get \pi and SNP density from the neutral regions of the genome.
- Full genome is not available because of data ethics reasons, but all we need are the summary statistics.
- We can start with either equilibrium or the 2-epoch model, \mu and standard human recombination map from litetature. Does it fit the \pi and SNP density of the data?
- Simulate a region of 1 Mb. This VCF with the candidate selected regions of interest is available to us.
- Train models, then apply to the STC1 region.
- Support conclusions with some other evidence. Especially if we say it's a soft sweep.
- If it's a simple application of my method, probably not a thesis chapter. If I go above and beyond, it might be.

**** Meeting [2021-11-11 Thu 10:30]
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-03 Thu 18:25
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_OLPATH: PhD/Turkana selection/Meetings
:END:

- Sweep in STC1 with circa 80% frequency.
- What is selection strength?
- Is the sweep hard or soft?
- Estimate sweep age if it's straightforward, but not necessary if it's a lot of trouble. I think it's a lot of trouble because it'd require assuming some model for a sweep that's both incomplete and old.
- Region is around 50kb, and we can try some different 50kb windows and see if results are different.
- Timeline: weeks to months.
- Paper is a selection story with scan, functional validation, and hopefully some inferences about the mode and strength of selection. That's where we come in.
- Paper about the specific STC1 locus for desert/dehydration adaptation. But also super pleiotropic.
- Paper will introduce entire dataset of 301 genomes.
- No cultural evolution, which was the original purpose of the first 100 high-quality genomes.
  
  



**** Discussion with Philipp [2021-11-05 Fri]
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-03 Thu 18:25
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_OLPATH: PhD/Turkana selection/Meetings
:END:

They want to do more sweep analysis on a single strong sweep candidate. They're primarily interested in the timing of it. But we can do sel. strength and sweep mode. Especially if we can show it's a likely soft sweep, from SGV or RNM.

Our job:
- Simulate.
  - What demographic model?
  - Any other human-specific simulation challenges?
  - As a first step, doing equilibrium demography and constant size is a good idea.
  - Try to fit pi.
  - Try to fir \mu/r.
  - Maybe we don't even need to rescale, since they have N_e = 10,000.
  - We can use recombination rates specific to the region. Use other populations as reference. Is it a uniform recombination rate? How much variation is there in local rec. rate in a region of circa 1Mb?
  - There's a chance they have a recombination map for their population, but that'd be biased because there's this strong sweep signal in the middle increasing LD.
  - What's the variation in rec. rate between African populations?
  - Do you see this sweep in other human populations?
- Train.
  - What machine learning model?
- Apply.
  - Support conclusions with some other evidence. Especially if we say it's a soft sweep.

Concerns:
- If it's a simple application of my method, probably not a thesis chapter. If I go above and beyond, it might be.
- What are their time constraints? Are they in a super hurry?
- What happened to Melissa Wilson that simulating Turkana in SLiM didn't work?

*** 2022-03-07 Monday
**** DONE Turkana with more simulated data [13/13]
CLOSED: [2022-03-07 Mon 17:12]
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-09 Wed 12:27
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_OLPATH: PhD/Turkana project/Turkana, more confident predictions
:END:
- State "DONE"       from "DOING"      [2022-03-07 Mon 17:12]
- State "DOING"      from "TODO"       [2022-03-04 Fri 18:12]
- State "DOING"      from "DONE"       [2022-03-02 Wed 08:15]
- State "DONE"       from "DOING"      [2022-03-01 Tue 23:58]
- State "DOING"      from "TODO"       [2022-03-01 Tue 20:00]
- State "SOON"       from              [2022-02-21 Mon 11:52]
- [X] Download simulated data. (In progress.)
- [X] Rename local simulations to =simple=.
- [X] Add rule to make empirical inference replicates.
- [X] Add notebook to make empirical inference replicates.
- [X] Run with =-n= to make sure everything still works.
- [X] Run everything with the new data and =-c8=.
- [X] Git update.
- [X] As it runs, check that the data reports match up. (Ready at 10:30?)
- [X] Try with =-n= to see what the current jobs are.
- [X] Implement replicates of everything, from the =all= rule down, using the same data.
- [X] Try with =-n= to confirm the jobs.
- [X] Run everything with 20,000 data points per sweep.
- [X] Confirm Everything ran okay.

*** 2022-03-08 Tuesday
**** DONE Gather/plot results of Turkana replicate inferences [14/14]
CLOSED: [2022-03-08 Tue 17:19]
:PROPERTIES:
:ARCHIVE_TIME: 2022-03-09 Wed 12:29
:ARCHIVE_FILE: ~/pessoal/essencial/agenda.org
:ARCHIVE_CATEGORY: PhD
:ARCHIVE_TODO: DONE
:ARCHIVE_OLPATH: PhD/Turkana project/Turkana further analyses
:END:

- State "DONE"       from "DOING"      [2022-03-08 Tue 17:19]
- State "DOING"      from "TODO"       [2022-03-08 Tue 12:34]
- [X] See if you can consolidate plotting and inference workfLows.
- [X] Tables/summaries of empirical inferences.
- [X] Copy rule from Drosophila.
- [X] Adjust paths in new rule.
- [X] Copy script/notebook from Drosophila.
- [X] Adjust script/notebook as necessary.
- [X] Make sure it works with =-n=.
- [X] Run it.
- [X] Check that results make sense.
- [X] Save "statistics" files with an index.
- [X] Move rule to the inference workflow.
- [X] Move notebook to the inference folder.
- [X] Re-run.
- [X] Git update.

